test/gtest/unsortedsegmentsum.hpp

```diff
@@ -0,0 +1,214 @@
+    std::vector<size_t> dims;
+    size_t num_segments;
+
+    friend std::ostream& operator<<(std::ostream& os, const UnsortedSegmentSumTestCase& tc)
+    {
+        return os << " dims:" << tc.dims << " num_segments:" << tc.num_segments;
+    }
+};
+
+inline std::vector<UnsortedSegmentSumTestCase> UnsortedSegmentSumTestConfigs()
+{ // n c d h w lr momentum dampening weightDecay nesterov momentumInitialized
+    return {
+        {{50, 10}, 10},
+        {{50, 10, 20}, 10},
+        {{50, 10, 20, 30}, 10},
+        {{50, 10, 20, 30, 4}, 10},
+    };
+}
+
+template <typename T = float>
+struct UnsortedSegmentSumTestFwd : public ::testing::TestWithParam<UnsortedSegmentSumTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle             = get_handle();
+        UnsortedSegmentSum_config = GetParam();
+
+        num_segments   = UnsortedSegmentSum_config.num_segments;
+        auto gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+        auto gen_segment_ids = [this](auto...) {
+            return prng::gen_descreet_uniform_sign<int>(0, num_segments);
+        };
+
+        auto dims        = UnsortedSegmentSum_config.dims;
+        auto output_dims = dims;
+        output_dims[0]   = num_segments;
+
+        segment_ids = tensor<int>{dims[0]}.generate(gen_segment_ids);
+        input       = tensor<T>{dims}.generate(gen_value);
+        output      = tensor<T>{output_dims};
+        ref_output  = tensor<T>(output);
+
+        std::fill(output.begin(), output.end(), 0);
+        std::fill(ref_output.begin(), ref_output.end(), 0);
+
+        input_dev       = handle.Write(input.data);
+        output_dev      = handle.Write(output.data);
+        segment_ids_dev = handle.Write(segment_ids.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        cpu_UnsortedSegmentSum_forward<T, int>(input, ref_output, segment_ids, num_segments);
+        miopenStatus_t status = miopenStatusSuccess;
+
+        status = miopen::UnsortedSegmentSum::UnsortedSegmentSumForward(handle,
+                                                                       input.desc,
+                                                                       input_dev.get(),
+                                                                       output.desc,
+                                                                       output_dev.get(),
+                                                                       segment_ids.desc,
+                                                                       segment_ids_dev.get());
+        ASSERT_EQ(status, miopenStatusSuccess);
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+        auto error       = miopen::rms_range(ref_output, output);
+
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
```

Comment:
`range_distance` check is already covered in `rms_range` so I think this line is unnecessary. Please refer to `rms_range`'s implementation

Reply:
please check my comment in your PR